Modules/_operator.c

```diff
@@ -1230,6 +1233,242 @@ static PyType_Spec itemgetter_type_spec = {
+[clinic start generated code]*/
+
+static PyObject *
+itemtuplegetter_new_impl(PyTypeObject *type, PyObject *itbl,
+                         PyObject *defaultitbl)
+/*[clinic end generated code: output=37d6a97f69ed1fc1 input=62951fc0246b99f0]*/
+
+{
+    itemtuplegetterobject *itg;
+    PyObject *items = NULL, *defaults = NULL;
+    Py_ssize_t nitems, ndefaults;
+
+    items = PySequence_Tuple(itbl);
+    if (items == NULL) {
+        Py_XDECREF(items);
+        return NULL;
+    }
+
+    nitems = PyTuple_GET_SIZE(items);
+
+    if (defaultitbl == Py_None) {
+        defaults = PyTuple_New(0);
+    } else {
+        defaults = PySequence_Tuple(defaultitbl);
+        if (defaults == NULL) {
+            Py_XDECREF(defaults);
+            return NULL;
+        }
+    }
+    ndefaults = PyTuple_GET_SIZE(defaults);
+
+    _operator_state *state = _PyType_GetModuleState(type);
+    /* create itemtuplegetterobject structure */
+    itg = PyObject_GC_New(itemtuplegetterobject, (PyTypeObject *) state->itemtuplegetter_type);
+    if (itg == NULL) {
+        return NULL;
+    }
+
+    itg->items = Py_NewRef(items);
+    itg->defaults = Py_NewRef(defaults);
+    itg->nitems = nitems;
+    itg->ndefaults = ndefaults;
+
+    PyObject_GC_Track(itg);
+    return (PyObject *)itg;
+}
+
+static int
+itemtuplegetter_clear(itemtuplegetterobject *itg)
+{
+    Py_CLEAR(itg->items);
+    Py_CLEAR(itg->defaults);
+    return 0;
+}
+
+static void
+itemtuplegetter_dealloc(itemtuplegetterobject *itg)
+{
+    PyTypeObject *tp = Py_TYPE(itg);
+    PyObject_GC_UnTrack(itg);
+    (void)itemtuplegetter_clear(itg);
+    tp->tp_free(itg);
+    Py_DECREF(tp);
+}
+
+static int
+itemtuplegetter_traverse(itemtuplegetterobject *itg, visitproc visit, void *arg)
+{
+    Py_VISIT(Py_TYPE(itg));
+    Py_VISIT(itg->items);
+    Py_VISIT(itg->defaults);
+    return 0;
+}
+
+static PyObject *
+itemtuplegetter_call(itemtuplegetterobject *itg, PyObject *args, PyObject *kw)
+{
+    assert(PyTuple_CheckExact(args));
+    if (!_PyArg_NoKeywords("itemtuplegetter", kw))
+        return NULL;
+    if (!_PyArg_CheckPositional("itemtuplegetter", PyTuple_GET_SIZE(args), 1, 1))
+        return NULL;
+    return itemtuplegetter_call_impl(itg, PyTuple_GET_ITEM(args, 0));
+}
+
+static PyObject *
+itemtuplegetter_call_impl(itemtuplegetterobject *itg, PyObject *obj)
+{
+    PyObject *result;
+    Py_ssize_t nitems=itg->nitems, ndefaults=itg->ndefaults;
+
+    result = PyTuple_New(nitems);
+    if (result == NULL)
+        return NULL;
+
+    if (nitems == 0)
+        return result;
+
+    Py_ssize_t i = 0;
+
+    if (ndefaults) {
+        Py_ssize_t min_items = ndefaults < nitems ? ndefaults : nitems;
+
+        for (i=0 ; i < min_items; i++) {
+            PyObject *item, *val, *found;
+
+            item = PyTuple_GET_ITEM(itg->items, i);
+            found = PyObject_GetItem(obj, item);
+            val = (found == NULL) ? PyTuple_GET_ITEM(itg->defaults, i) : found;
```

Comment:
would be clearer to do a statement instead of ternary.

Also, need to `Py_INCREF` in case of `default`